test/mlir/krnl/krnl_getref_lowering.mlir

```diff
@@ -66,17 +63,15 @@ func @test_getref_lowering_dynamic(%arg0: memref<2x2xf32>) -> memref<2x?xf32> {
   // CHECK: [[ALLOC_MEMREF_3:%.+]] = llvm.insertvalue [[TYPED_ALLOC]], [[ALLOC_MEMREF_2]][1]
   // CHECK: [[CONST_0:%.+]] = llvm.mlir.constant(0 : index) : !llvm.i64
   // CHECK: [[ALLOC_MEMREF_4:%.+]] = llvm.insertvalue [[CONST_0]], [[ALLOC_MEMREF_3]][2]
-  // CHECK: [[CONST_1:%.+]] = llvm.mlir.constant(1 : index) : !llvm.i64
-  // CHECK: [[MUL2:%.+]] = llvm.mul %23, %[[C5_INDEX]] : !llvm.i64
   // CHECK: [[ALLOC_MEMREF_5:%.+]] = llvm.insertvalue %[[C10_INDEX]], [[ALLOC_MEMREF_4]][3, 0]
-  // CHECK: [[ALLOC_MEMREF_6:%.+]] = llvm.insertvalue [[MUL2]], [[ALLOC_MEMREF_5]][4, 0]
-  // CHECK: [[ALLOC_MEMREF_7:%.+]] = llvm.insertvalue %[[C5_INDEX]], [[ALLOC_MEMREF_6]][3, 1]
-  // CHECK: [[ALLOC_MEMREF_8:%.+]] = llvm.insertvalue [[CONST_1]], [[ALLOC_MEMREF_7]][4, 1]
+  // CHECK: [[ALLOC_MEMREF_6:%.+]] = llvm.insertvalue %[[C5_INDEX]], [[ALLOC_MEMREF_5]][3, 1]
+  // CHECK: [[ALLOC_MEMREF_7:%.+]] = llvm.insertvalue %[[C5_INDEX]], [[ALLOC_MEMREF_6]][4, 0]
+  // CHECK: [[ALLOC_MEMREF_8:%.+]] = llvm.insertvalue %[[C1_INDEX]], [[ALLOC_MEMREF_7]][4, 1]
```

Comment:
Why is there a change in the order of these inserts?

Reply:
I don't know the cause of this one, but the results looked correct. I wasn't sure what the project standards are for changes like this, and there were a number of these swaps. I can dive into the pass if you would find that helpful.

Reply:
I find it a little strange but it could be a consequence of some changes in the way MemRefs are lowered by MLIR. Just wanted to sanity check this.